### ISSUE TITLE
Consider Azure VM scale sets for Workspaces

### ISSUE THREAD

**Issue by jemrobinson** [alan-turing-institute/data-safe-haven#1859]
## :white_check_mark: Checklist

- [x] I have searched open and closed issues for duplicates.
- [x] This is a request for a new feature in the Data Safe Haven or an upgrade to an existing feature.
- [x] The feature is still missing in the [latest version](https://github.com/alan-turing-institute/data-safe-haven/releases).
- [x] I have read through the [documentation](https://alan-turing-institute.github.io/data-safe-haven/).
- [x] This isn't an open-ended question (open a [discussion](https://github.com/alan-turing-institute/data-safe-haven/discussions) if it is).

## :strawberry: Suggested change

We could use Azure VM Scale Sets (https://learn.microsoft.com/en-us/azure/virtual-machine-scale-sets/flexible-virtual-machine-scale-sets-portal) to automatically scale the number of deployed workspaces based on demand.

## :steam_locomotive: How could this be done?

As each VMSS is limited to a single SKU we would probably want one for CPU-enabled and one for GPU-enabled machines.

**Comment by JimMadge** [comment]
Do you know how these work? I'm thinking about configuring networking and NFS/samba.

Can we run configuration at deploy time? Can we have a pre-configured image which all VMs are deployed from?

Tempted to move this to a later milestone.

**Comment by jemrobinson** [comment]
Yes, I think it probably should be moved to a later milestone. Just wanted to make sure we were thinking about it and moving it deliberately.
